jerry-core/ecma/base/ecma-globals.h

```diff
@@ -729,8 +729,10 @@ typedef enum
 #endif /* JERRY_ESNEXT */
 #if JERRY_BUILTIN_TYPEDARRAY
   ECMA_OBJECT_CLASS_ARRAY_BUFFER, /**< Array Buffer (ECMAScript v6, 24.1) */
-  ECMA_OBJECT_CLASS_SHARED_ARRAY_BUFFER, /**< Shared Array Buffer (ECMAScript v11 24.2) */
 #endif /* JERRY_BUILTIN_TYPEDARRAY */
+#if JERRY_BUILTIN_SHAREDARRAYBUFFER
+  ECMA_OBJECT_CLASS_SHARED_ARRAY_BUFFER, /**< Shared Array Buffer (ECMAScript v11 24.2) */
+#endif /* JERRY_BUILTIN_SHAREDARRAYBUFFER */
```

Comment:
Shouldn't this be inside the TypedArray guard? In the file above the `ECMA_OBJECT_CLASS_SHARED_ARRAY_BUFFER` is guarded by both in the switch case. (Well if the correctly check the cross-dependency in the config.h then this should not matter).